src/lib.rs

```diff
@@ -121,4 +121,246 @@ impl zed::Extension for ScalaExtension {
     }
 }
 
+use zed_extension_api::{DebugAdapterBinary, DebugTaskDefinition, StartDebuggingRequestArgumentsRequest, Worktree};
+
+impl ScalaExtension {
+    fn get_metals_dap_path(&self, worktree: &zed::Worktree) -> Result<String> {
+        worktree
+            .which("metals-dap")
+            .ok_or_else(|| "Metals DAP server (metals-dap) must be installed manually. Please ensure it is in your PATH.".to_string())
+    }
+}
+
+impl zed::Extension for ScalaExtension {
+    fn new() -> Self {
+        Self
+    }
+
+    fn language_server_command(
+        &mut self,
+        _language_server_id: &zed::LanguageServerId,
+        worktree: &zed::Worktree,
+    ) -> Result<zed::Command> {
+        let path = worktree
+            .which("metals")
+            .ok_or_else(|| "Metals must be installed manually. Recommended way is to install coursier (https://get-coursier.io/), and then run `cs install metals`.".to_string())?;
+
+        let arguments = LspSettings::for_worktree("metals", worktree)
+            .map(|lsp_settings| {
+                lsp_settings
+                    .binary
+                    .and_then(|binary| binary.arguments)
+                    // If no arguments are provided, default to enabling the HTTP server.
+                    .unwrap_or(vec!["-Dmetals.http=on".to_string()])
+            })
+            .unwrap_or_default();
+
+        Ok(zed::Command {
+            command: path,
+            args: arguments,
+            env: worktree.shell_env(),
+        })
+    }
+
+    fn language_server_initialization_options(
+        &mut self,
+        _language_server_id: &zed_extension_api::LanguageServerId,
+        worktree: &zed_extension_api::Worktree,
+    ) -> Result<Option<serde_json::Value>> {
+        let initialization_options = LspSettings::for_worktree("metals", worktree)
+            .map(|lsp_settings| lsp_settings.initialization_options.clone());
+
+        initialization_options
+    }
+
+    fn language_server_workspace_configuration(
+        &mut self,
+        _language_server_id: &zed::LanguageServerId,
+        worktree: &zed::Worktree,
+    ) -> Result<Option<serde_json::Value>> {
+        let settings = LspSettings::for_worktree("metals", worktree)
+            .ok()
+            .and_then(|lsp_settings| lsp_settings.settings.clone())
+            .unwrap_or_default();
+
+        Ok(Some(serde_json::json!({
+            "metals": settings
+        })))
+    }
+
+    fn label_for_completion(
+        &self,
+        _language_server_id: &zed_extension_api::LanguageServerId,
+        completion: Completion,
+    ) -> Option<zed_extension_api::CodeLabel> {
+        let prefix = match completion.kind? {
+            CompletionKind::Method | CompletionKind::Function => "def ",
+            CompletionKind::Constructor
+            | CompletionKind::Class
+            | CompletionKind::Interface
```

Comment:
Don't know all the details of this integration, but for `CompletionKind::Interface` perhaps we should use `trait` label instead? :slightly_smiling_face: